state/datastream.go

```diff
@@ -531,6 +588,10 @@ func GenerateDataStreamFile(ctx context.Context, streamServer *datastreamer.Stre
 						BlockGasLimit:   l2Block.BlockGasLimit,
 					}
 
+					streamL2BlockEnd := &datastream.L2BlockEnd{
```

Comment:
Maybe this code can be added inside the if statement in line 691, since streamL2BlockEnd is used only u ds version i >= 4